src/common/ad_mem.c

```diff
@@ -2790,9 +2805,10 @@ ad_root(struct ad_blob_handle bh, size_t size)
 
 /** Query base pointer */
 void *
-ad_get_base_ptr(struct ad_blob_handle bh)
+ad_base(struct ad_blob_handle bh)
 {
 	struct ad_blob	*blob = bh.bh_blob;
 
-	return ad_addr2ptr(bh, blob_addr(blob));
+	D_ASSERT((uintptr_t)ad_addr2ptr(bh, blob_addr(blob) == (uintptr_t)blob->bb_mmap));
```

Comment:
sorry here with a bug, should be "D_ASSERT((uintptr_t)ad_addr2ptr(bh, blob_addr(blob)) == (uintptr_t)blob->bb_mmap);" will fix it when refresh it next time.